actors/builtin/multisig/multisig_actor.go

```diff
@@ -83,24 +72,26 @@ type ProposeReturn struct {
 func (a *MultiSigActor) Propose(rt vmr.Runtime, params *ProposeParams) *ProposeReturn {
 	rt.ValidateImmediateCallerType(builtin.CallerTypesSignable...)
 	callerAddr := rt.ImmediateCaller()
-	a._rtValidateSignerOrAbort(rt, callerAddr)
-
-	h, st := a.State(rt)
-	txnID := st.NextTxnID
-	st.NextTxnID += 1
-
-	st.PendingTxns = setPendingTxn(rt, st.PendingTxns, txnID, MultiSigTransaction{
-		To:       params.To,
-		Value:    params.Value,
-		Method:   params.Method,
-		Params:   params.Params,
-		Approved: []addr.Address{callerAddr},
-	})
 
-	UpdateRelease_MultiSig(rt, h, st)
+	var txnID TxnID
+	var st MultiSigActorState
+	rt.State().Transaction(&st, func() interface{} {
+		a.validateAuthorizedParty(rt, &st, callerAddr)
+		txnID := st.NextTxnID
+		st.NextTxnID += 1
+
+		st.PendingTxns = setPendingTxn(rt, st.PendingTxns, txnID, MultiSigTransaction{
+			To:       params.To,
+			Value:    params.Value,
+			Method:   params.Method,
+			Params:   params.Params,
+			Approved: []addr.Address{},
```

Comment:
Do we need to add `calledAddr` to this slice as was done before?

Reply:
This is an explicit change, because approveTransaction will add it. Otherwise it would have aborted always.